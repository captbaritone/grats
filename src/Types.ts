import type { GraphQLResolveInfo } from "graphql";

/** @gqlScalar */
export type Float = number;
/** @gqlScalar */
export type Int = number;
/** @gqlScalar */
export type ID = string;

/** @gqlInfo */
export type GqlInfo = GraphQLResolveInfo;
