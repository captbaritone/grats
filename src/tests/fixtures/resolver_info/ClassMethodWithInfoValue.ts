import { GqlInfo } from "../../../Types";

/** @gqlType */
export class SomeType {
  /** @gqlField greeting */
  greetz(info: GqlInfo): string {
    return "Hello";
  }
}
