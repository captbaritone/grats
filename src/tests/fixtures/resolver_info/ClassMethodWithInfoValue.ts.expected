-----------------
INPUT
----------------- 
import { GqlInfo } from "../../../Types";

/** @gqlType */
export class SomeType {
  /** @gqlField greeting */
  greetz(info: GqlInfo): string {
    return "Hello";
  }
}

-----------------
OUTPUT
-----------------
-- SDL --
type SomeType {
  greeting: String @metadata(name: "greetz")
}
-- TypeScript --
import { GraphQLSchema, GraphQLObjectType, GraphQLString } from "graphql";
export function getSchema(): GraphQLSchema {
    const SomeTypeType: GraphQLObjectType = new GraphQLObjectType({
        name: "SomeType",
        fields() {
            return {
                greeting: {
                    name: "greeting",
                    type: GraphQLString,
                    resolve(source, _args, _context, info) {
                        return source.greetz(info);
                    }
                }
            };
        }
    });
    return new GraphQLSchema({
        types: [SomeTypeType]
    });
}
